examples/language-modeling/run_clm.py

```diff
@@ -0,0 +1,323 @@
+# coding=utf-8
+# Copyright 2018 The Google AI Language Team Authors and The HuggingFace Inc. team.
+# Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Fine-tuning the library models for causal language modeling (GPT, GPT-2, CTRL) on a text file or a dataset.
+"""
+
+
+import logging
+import math
+import os
+import sys
+from dataclasses import dataclass, field
+from typing import Optional
+
+from datasets import load_dataset
+
+import transformers
+from transformers import (
+    CONFIG_MAPPING,
+    MODEL_FOR_CAUSAL_LM_MAPPING,
+    AutoConfig,
+    AutoModelForCausalLM,
+    AutoTokenizer,
+    HfArgumentParser,
+    Trainer,
+    TrainingArguments,
+    default_data_collator,
+    set_seed,
+)
+from transformers.trainer_utils import is_main_process
+
+
+logger = logging.getLogger(__name__)
```

Comment:
Should we use `transformers`'s  library `logging` ? (cc @LysandreJik)

Reply:
No, for the script, we should use the regular one. @LysandreJik had a very long explanation of why that I don't remember.

Reply:
[Here it is.](https://github.com/huggingface/transformers/pull/7917#discussion_r508463451)

The gist of it is that imo the `transformer` `logging` utility should only be used to control the logging of the `transformers` module, not of the users' scripts directly, as it is not made for that and would lead to very weird behavior.

In my opinion the control of logging in a user script should contain both:

```py
import logging
from transformers import logging as hf_logging

hf_logging.set_verbosity_xxx()
logger = logging.getLogger(__name__)

# then do stuff with the logger without worrying about the HF logging which has already been managed before
logger.warn("xxx")
```